### DISCUSSION TITLE
ClusterRole controlled of two ArgoCD applications

### DISCUSSION THREAD

**Discussion by tiefseeruebe** [argoproj/argo-cd#17627] [Q&A] [answered]
I have a ClusterRole and a ClusterRoleBinding that is created by one application. Another application gets out of sync, because it claims to be "responsible" for the cluster role and binding created by the first application. So it would delete both, the cluster role and the binding if synced with the prune option. The first application has the same issue with a cluster role and binding created be the second application. How can I avoid this interference? Why does an application "believes" it is responsible for a given k8s object?

**Comment by jgwest** [comment, marked as answer]
Argo CD uses the `app.kubernetes.io/instance` annotation on  a resource to indicate which Argo CD Application deployed that resource. You can learn more about that [here](https://argo-cd.readthedocs.io/en/stable/user-guide/resource_tracking/), and other mechanisms are supported as well.

In general, K8s resources should not be shared between Argo CD `Application`s, but it sounds like in your case it may be an issue with conflicting annotations.

**Comment by tiefseeruebe** [comment]
Thanks jgwest for your answer, that helped me to find the issue. I had a look the documentation you provided and on my ClusterRoles. Actually my ArgoCD uses not the label but the annotation for the resource tracking in `metadat
a.annotations.argocd.argoproj.io/tracking-id` with the format `<application-name>:<kind>:<name-space>/<cluster-role-name>`. This seems to be the reason that resources w/o namespace are not considered to belong to the correct application. I think we are going to rename the applications to avoid this interference. This way the assignment should be clear.
